src/main/java/juja/microservices/keepers/service/KeepersService.java

```diff
@@ -42,14 +46,35 @@
         return result;
     }
 
-    public String addKeeper(KeeperRequest keeperRequest){
+    public List<String> inactiveKeeper(KeeperRequest keeperRequest) {
+        String uuid = keeperRequest.getUuid();
+        String direction = keeperRequest.getDirection();
+        String form = keeperRequest.getFrom();
+        logger.debug("Service.deleteKeeper after in, parameters: {}", keeperRequest.toString());
+        if (keepersRepository.findOneActive(form) == null) {
+            logger.warn("Keeper '{}' tried to set inactive 'Keeper' '{}' but he's not an active Keeper", form, uuid);
+            throw new KeeperAccessException("Only active keeper could set inactive another keeper");
+        }
+        Keeper keeper = keepersRepository.findOneByUUIdAndDirectionIsActive(uuid, direction);
+        if (keeper == null) {
+            logger.warn("Keeper with uuid '{}' and direction '{}' is't exist or inactive", uuid, direction);
+            throw new KeeperNonexistentException("Keeper with uuid " + uuid + " and direction " + direction
+                    + " is't exist or inactive");
+        }
+        keeper.setDismissDate(LocalDateTime.now());
+        List<String> ids = Collections.singletonList(keepersRepository.inactive(keeper));
+        logger.info("'Keeper' updated , with uuid {}, from user '{}'", uuid, form);
+        return ids;
+    }
+
+    public String addKeeper(KeeperRequest keeperRequest) {
         logger.debug("Service.addKeeper after in, parameters: {}", keeperRequest.toString());
-        if(keepersRepository.findOneByUUId(keeperRequest.getFrom()) == null){
+        if (keepersRepository.findOneByUUId(keeperRequest.getFrom()) == null) {
```

Comment:
If you check in the method above that only active keeper can inactivate another keeper then we have to check in the same way during additing new keeper. Please change addKeeper and delete extra method `findOneByUuid` from repository.